50-dhole/technical_specification.md

```diff
@@ -240,10 +240,141 @@ B. For merging relation properties:
   ```
 
 ### Refactor aggregate transformation:
-TBD.
+
+The former `aggregate` transformation shall be replaced by a mixture of existing
+and to be written transformations. The following new transformations shall be
+implemented:
+
+#### Transformation 1: Add Content Property
+
+A transformation that enables the insertion of a subschema into a new property
+of any object within the current content schema of a class, including an initial
+value. The schema will be modified by inserting the subschema at the specified path. The data will be modified by inserting the default value at the specified path. The user may control whether or not the newly added property is added to the parent property's `required` list.
+
+##### Configuration
+
+```yaml
+# Generic Example, defaults shown for optional parameters
+- class_name: MyClass
```

Comment:
Should we stay with arrays of instructions for our transformation steps? I am also happy to defer that decision to later.

Reply:
As discussed, will keep the current representation.